ansible_runner/utils/capacity.py

```diff
@@ -0,0 +1,17 @@
+import multiprocessing
+import resource
+
+
+def get_cpu_capacity():
+    # `multiprocessing` info: https://docs.python.org/3/library/multiprocessing.html
+    forkcpu = 4
+    cpu_capacity = multiprocessing.cpu_count() * forkcpu
+    return cpu_capacity
+
+
+def get_mem_capacity():
+    # `resource` info: https://docs.python.org/3/library/resource.html
+    byte_denom = 1024
+    mem = resource.getrusage(resource.RUSAGE_SELF).ru_maxrss 
```

Comment:
I think this is probably an accurate way to get the memory available. I'm doing some testing to confirm.

Reply:
This one is more tricky. I haven't been able to find a way to detect limited memory from within the container. I imagine the most accurate way would be inspecting the `cgroup` from `/proc/1/cgroup`, but I didn't find anything definitive.

Reply:
For some background, this would replace the logic we have in AWX:

for CPU:

https://github.com/ansible/awx/blob/747231d3500d86ba07006a6927498d8ec31bf930/awx/main/utils/common.py#L716

and for memory:

https://github.com/ansible/awx/blob/747231d3500d86ba07006a6927498d8ec31bf930/awx/main/utils/common.py#L748

I've dug enough into the psutil implementation to say that it just gives the value from `/proc/meminfo` (for Linux). We can't use the old logic here, because we _just_ removed psutil as a dependency of ansible-runner, with much enthusiasm for it.

I only mean to say - if our implementation is _as bad_ as the old psutil one, it's not a regression. Doesn't mean we can't improve, but that's where we're coming from, because this is for AWX capacity-based job allocation.